src/services/server/user-repository/supabase-user-repository.ts

```diff
@@ -47,6 +50,61 @@ export const SupabaseUserRepository = inject(
         throw new ServerError('Failed to parse user data.', 400);
       }
     }
+    /**
+     * @updateRegisterToVoteAction
+     * @param id - User's id to get user
+     */
+    async updateRegisterToVoteAction(userId: string): Promise<void> {
+      const supabase = this.createSupabaseClient();
+
+      const { error: challengerUpdateError } = await supabase
+        .from('completed_actions')
+        .update({
+          register_to_vote: true,
+        })
+        .eq('user_id', userId);
+
+      if (challengerUpdateError) {
+        throw new Error(challengerUpdateError.message);
+      }
+    }
+    /**
+     * @awardUserBadge
+     * @param id - User's id to get user
+     */
+    async awardVoterRegistrationActionBadge(
+      id: string,
+      badges: Badge[],
+    ): Promise<void> {
+      const supabase = this.createSupabaseClient();
+
+      let found = false;
+      const actionBadges = badges.filter(obj => {
+        if ((obj as ActionBadge).action === 'voterRegistration') {
+          found = true;
+        }
+        return (obj as ActionBadge).action !== undefined;
+      });
+
+      if (badges.length >= 8 || found) {
+        //should we be throwing an error here? throw new Error("can not award user a badge")
+        return;
+      }
```

Comment:
I don't think we should throw. I think what would be better would be to have a separate method that checks if a badge should be awarded, and then only call this method if that method returns true.

Reply:
Gotcha! Yep, I made a private method like this: 

private canAwardBadge = (user: User): Boolean => {
        if (
          user.badges.length >= 8 ||
          user.completedChallenge ||
          user.completedActions.registerToVote
        ) {
          return false;
        }
        return true
      },